Update references to shared objects as supported, link to related mat… (#4135)
MystenLabs/sui: doc/src/build/move/sui-move-library.md

```
After this call, `obj` becomes immutable which means it can never be mutated or deleted. This process is also irreversible: once an object is frozen, it will stay frozen forever. An immutable object can be used as reference by anyone in their Move call.

### Share an object
To make an object `obj` shared and mutable, one can call:
```
transfer::share_object(obj);
```

> After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be deleted, transferred or embedded in another object as a field. For an example, see [Shared Object](https://examples.sui.io/basics/shared-object.html#shared-object) on https://examples.sui.io/.

Reviewer: ```suggestion
After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be transferred or embedded in another object as a field. For more detail, see the [shared objects](../../learn/build/objects.md#shared) documentation.
```

- I moved the more thorough description of shared objects to the "objects" section and simply linked to that description here
- We will soon (but do not yet) support shared object deletion, so I think we can avoid talking about deletion for now + explicitly mention that is supported once it lands.

Author: Thanks, Sam!

> Shared mutable objects can be powerful in that they make programming a lot simpler in many cases. However, shared objects are also more expensive to use: they require a full sequencer (a.k.a. a [consensus engine](../../learn/architecture/consensus.md) to order the transactions that touch the shared object, which means longer latency/lower throughput and higher gas cost. One can see the difference in the two programming schemes by looking at the two implementations of TicTacToe: [No Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/tic_tac_toe.move) vs. [Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/shared_tic_tac_toe.move).

Reviewer: ```suggestion
```
